src/core_plugins/input_control_vis/public/components/vis/input_control_vis.js

```diff
@@ -44,8 +44,10 @@ export class InputControlVis extends Component {
               selectedOptions={control.value}
               disableMsg={control.isEnabled() ? null : control.disabledReason}
               multiselect={control.options.multiselect}
+              dynamicOptions={control.options.dynamicOptions}
               controlIndex={index}
               stageFilter={this.props.stageFilter}
+              fetchOptions={async (query) => { return await this.props.refreshControl(index, query); }}
             />
```

Comment:
In this case, the caller of `fetchOptions` is doing an await, so you can make this `query => this.props.refreshControl(index, query)` since the caller is handling a promise anyway.

Reply:
Er. Also, I wonder if we can get rid of `props.refreshControl` altogether, and just do `query => control.fetch(query)` here, and allow the base Control to intelligently refresh when a query completes? Not sure. I don't know enough about this.

Reply:
There is no good way to update control in this component since it's state is passed in as props.